repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilter.scala

```diff
@@ -19,7 +19,193 @@
  */
 package org.openrepose.powerfilter
 
-class ReposeFilter {
-  //todo: Set the start time attribute (header?) before invoking the filter chain, it didn't make any sense to be setting it everytime we entered the old filter chain
-  //todo: Conditionally add a trace request mapping to the logging MDC if a trace header is present on the request
+import java.net.{URI, URISyntaxException}
+import java.util.{Optional, UUID}
+
+import com.codahale.metrics.MetricRegistry
+import io.opentracing.Tracer
+import io.opentracing.tag.Tags
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.HttpServletResponse._
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import org.apache.commons.lang3.StringUtils
+import org.openrepose.commons.utils.http.CommonHttpHeader.{REQUEST_ID, TRACE_GUID, VIA}
+import org.openrepose.commons.utils.http.PowerApiHeader.TRACE_REQUEST
+import org.openrepose.commons.utils.io.BufferedServletInputStream
+import org.openrepose.commons.utils.io.stream.LimitedReadInputStream
+import org.openrepose.commons.utils.logging.apache.format.stock.ResponseTimeHandler.START_TIME_ATTRIBUTE
+import org.openrepose.commons.utils.logging.{TracingHeaderHelper, TracingKey}
+import org.openrepose.commons.utils.opentracing.ScopeHelper.{closeSpan, startSpan}
+import org.openrepose.commons.utils.scala.TryWith
+import org.openrepose.commons.utils.servlet.http.ResponseMode.MUTABLE
+import org.openrepose.commons.utils.servlet.http.{HttpServletRequestWrapper, HttpServletResponseWrapper}
+import org.openrepose.core.services.reporting.metrics.MetricsService
+import org.openrepose.core.services.uriredaction.UriRedactionService
+import org.openrepose.core.spring.ReposeSpringProperties
+import org.openrepose.core.systemmodel.config.{Filter => _}
+import org.openrepose.nodeservice.containerconfiguration.ContainerConfigurationService
+import org.openrepose.nodeservice.response.ResponseHeaderService
+import org.openrepose.powerfilter.ReposeFilter._
+import org.slf4j.{LoggerFactory, MDC}
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.web.filter.DelegatingFilterProxy
+
+import scala.util.{Failure, Success}
+
+@Named("reposeFilter")
+class ReposeFilter @Inject()(@Value(ReposeSpringProperties.NODE.NODE_ID) nodeId: String,
+                             @Value(ReposeSpringProperties.CORE.REPOSE_VERSION) reposeVersion: String,
+                             optMetricsService: Optional[MetricsService],
+                             reposeFilterLoader: ReposeFilterLoader,
+                             containerConfigurationService: ContainerConfigurationService,
+                             tracer: Tracer,
+                             uriRedactionService: UriRedactionService,
+                             responseHeaderService: ResponseHeaderService
+                            )
+  extends DelegatingFilterProxy {
```

Comment:
Is this still needed with the removal of the war?

Reply:
Moved this to just use the `Filter`.